response.go

```diff
@@ -33,6 +33,9 @@ type ProcessingResult interface {
 	// GetStatusCode returns the status code of the response
 	GetStatusCode() int
 
+	// SetStatusCode allows setting the status code of the response
+	SetStatusCode() int
```

Comment:
Both `Response` and `ResponseStream` get the `statusCode` as a parameter and doesn't return anything
```suggestion
	SetStatusCode(int)
```